Anagram.java

```diff
@@ -29,22 +28,63 @@ public static void main(String args[]) {
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		str1=preProcess(str1);
+		str2=preProcess(str2);
+		boolean check = true;
+		String str2temp = str2;
+		for (int i=0;i<str1.length();i++){
+			check = false;
+			str2= str2temp; 
+			for (int j=0;j<str2.length();j++) {
+				str2temp = "";
+				if (str1.charAt(i) == str2.charAt(j)) {
+					check = true;
+					for (int t=0;t<str2.length();t++){
+						if (t!=j) {
+							str2temp=str2temp + str2.charAt(t); 
+						}
+					}
+					j=str2.length();
+				}
+			}
+			if (check == false) {
+				i = str1.length();
+			}
+		}
+		return check;
+		
 	}
 	   
 	// Returns a preprocessed version of the given string: all the letter characters are converted
 	// to lower-case, and all the other characters are deleted, except for spaces, which are left
 	// as is. For example, the string "What? No way!" becomes "whatnoway"
-	private static String preProcess(String str) {
-		// Replace the following statement with your code
-		return str;
+	public static String preProcess(String str) {
+		String alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890 ";
+		String newString = "";
+		for (int i=0;i<str.length();i++) {
+			for (int j=0;j<alphabet.length();j++) {
```

Comment:
After reading the Anagram .. 
I have come to the realization that there are couple of tools that you dont understand fully... 
The main one is functions.
The second one is checking appearance with `str.indexOf(chr);`
- **if the string 'str' don't have 'chr' the function will return -1**; example "hello".indexOf('P'); // -1
- otherwise it will return the index of first appearance in string -> value will be in range [0, str.length()) 
-- Includes 0, not includes str.length()
examples
1. "hello".indexOf('h'); // 0
2. "hello".indexOf('l'); // 2

so if we will check with indexOf(chr) and use '!=' operator to compare to -1, we basically ask if str contains chr
I want you to go over to revisit some codes: 
a. Lecture 2-1, `DeleteVowels` question, slides: 27-29. (you also have code in Moodle) 
b.`isFirstCharInArgDigit`, `IsValidNumber` from WS2 (in Moodle)